README.md

```diff
@@ -1,6 +1,73 @@
 # Module Factory
+
 [![Build Status](https://github.com/gnosis/module-factory/actions/workflows/ci.yml/badge.svg)](https://github.com/gnosis/module-factory/actions)
 [![Coverage Status](https://coveralls.io/repos/github/gnosis/module-factory/badge.svg?branch=master)](https://coveralls.io/github/gnosis/module-factory?branch=master)
 
-This repository aims to make easier the deployment and track of Safe Modules using [Minimal Proxy Pattern](https://eips.ethereum.org/EIPS/eip-1167)
+The purpose of the Module Factory repository is to make the deployment of Safe Modules easier. 
+Applying the Minimal Proxy Pattern, this module reduces the gas cost and simplifies the track of deployed modules. 
+The Minimal Proxy Pattern has been used because the modules do not need to be upgradeable since a safe can deploy a new one. 
+It's worth mentioning that it costs roughly 5k additional gas for each transaction when using a proxy. Thus, after a certain number of transactions (~700) it would likely be cheaper to deploy the module from the constructor rather than the proxy.
```

Comment:
```suggestion
It's worth mentioning that it costs roughly 5k additional gas for each transaction when using a proxy.
Thus, after a certain number of transactions (~700) it would likely be cheaper to deploy the module from the constructor rather than the proxy.
```